### ISSUE TITLE
Add customToken option to firebase

### ISSUE THREAD

**Issue by viperfx** [redwoodjs/redwood#3491]
I currently use a custom firebase client because I utilize their signInWithCustomToken method.

```
      if (options.providerId == 'token' && options.token) {
        return firebaseAuth.signInWithCustomToken(auth, options.token)
      }
```

It would be great if we can add this option to the redwood firebase client.

**Comment by thedavidprice** [comment]
cc @doesnotexist

**Comment by doesnotexist** [comment]
Yeah, so the web/authclient side seems pretty straightforward.   As for the api-side decoder, would need to test it out a bit, because there are 2 ways you can generate the custom tokens, from the firebase docs:
"You can create a custom token with the Firebase Admin SDK, or you can use a third-party JWT library if your server is written in a language which Firebase does not natively support."

Would just want to make sure that the call to verifyIdToken in the decoder handles the third-party JWT case gracefully.

**Comment by viperfx** [comment]
Yep that's a good point. 

I currently only use firebase generated tokens for SSO and when I created this issue, I had actually meant only to support that. I don't think I've come across anyone who uses custom tokens yet.
